astropy/utils/masked/function_helpers.py

```diff
@@ -639,7 +639,7 @@ def quantile(a, q, axis=None, out=None, **kwargs):
     r, k = np.lib.function_base._ureduce(
         a, func=_masked_quantile, q=q, axis=axis, out=out, **kwargs
     )
-    return (r.reshape(k) if keepdims else r) if out is None else out
+    return (r.reshape(q.shape + k) if keepdims else r) if out is None else out
```

Comment:
This is what is done in numpy too (well, pre-1.25); silly to have missed that before.

Reply:
Is the numpy-dev fix that bad that it needs to be a separate PR?